src/modules/navigator/mission.cpp

```diff
@@ -996,16 +1006,29 @@ Mission::set_mission_items()
 				break;
 			}
 		}
+
+		if (_mission_item.nav_cmd == NAV_CMD_CONDITION_GATE) {
+			_mission_item.autocontinue = true;
+			_mission_item.time_inside = 0;
+		}
 	}
 
 	/*********************************** set setpoints and check next *********************************************/
 
 	position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
 
-	/* set current position setpoint from mission item (is protected against non-position items) */
-	if (new_work_item_type != WORK_ITEM_TYPE_PRECISION_LAND) {
-		mission_apply_limitation(_mission_item);
-		mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current);
+	if (item_contains_gate(_mission_item)) {
+		if (has_next_position_item) {
+			/* got next mission item, update setpoint triplet */
+			mission_item_to_position_setpoint(mission_item_next_position, &pos_sp_triplet->current);
+		}
```

Comment:
What happens in this `else`? We have a gate but no next position item? And does this ever happen?

Reply:
The position set point just stays what it was before, which is well-defined behavior.